### PR TITLE
Add more and document existing environment variables

### PR DESCRIPTION
**Pull request by jougs**
This is a replacement for #418, which adds the environment variables `SLI_PATH` for adding paths to the SLI search path and `NEST_MODULES` for loading modules on startup on the SLI level. It also adds a new documentation file that describes all environment variables that NEST reads.

I propose @obreitwi, @heplesser and @apeyser as reviewers.

### CODE REVIEW

File: extras/userdoc/md/documentation/environment-variables.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    startup using the `Install` SLI command. The path of the modules
+    has to be set using `NEST_MODULE_PATH`
+
+`NESTRCFILENAME`
+  : The name of NEST's configuration file (default: `~/.nestrc`). This
+    can be used to set different configurations for different
+    installations or to accomodate different machine setups.
+
+`SLI_PATH`
+  : A colon separated list of paths, in which NEST searches for SLI
```
Comment (apeyser) [inline extras/userdoc/md/documentation/environment-variables.md]: paths, in which --> paths in which

File: extras/userdoc/md/documentation/environment-variables.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    files. These are added to the search path.
+
+In addition to the NEST specific environment variables above, NEST
+also respects the following standard POSIX shell variables:
+
+`COLUMNS`
+  : A number of columns, to which NEST adapt the width of its textual
```
Comment (apeyser) [inline extras/userdoc/md/documentation/environment-variables.md]: columns, to which -> columns to which

**Comment by apeyser** [comment]
Some English lint, looks like it does the same as #418.
:+1:

**Comment by obreitwi** [comment]
Looks good to me. Thank you!

**Comment by jougs** [comment]
@apeyser: I've fixed the English.
@obreitwi: thanks!

**Pull request merged**